pkg/controllers/infracluster/infracluster_controller.go

```diff
@@ -222,63 +171,51 @@ func (r *InfraClusterController) reconcileInfraCluster(ctx context.Context, log
 	return ctrl.Result{}, nil
 }
 
-// ensureAWSCluster ensures the AWSCluster cluster object exists.
-func (r *InfraClusterController) ensureAWSCluster(ctx context.Context, log logr.Logger) (client.Object, error) {
-	target := &awsv1.AWSCluster{ObjectMeta: metav1.ObjectMeta{
-		Name:      r.Infra.Status.InfrastructureName,
-		Namespace: defaultCAPINamespace,
-	}}
-
-	// Checking whether InfraCluster object exists. If it doesn't, create it.
-
-	if err := r.Get(ctx, client.ObjectKeyFromObject(target), target); err != nil && !cerrors.IsNotFound(err) {
-		return nil, fmt.Errorf("failed to get InfraCluster: %w", err)
-	} else if err == nil {
-		return target, nil
-	}
-
-	log.Info(fmt.Sprintf("AWSCluster %s/%s does not exist, creating it", target.Namespace, target.Name))
+// ensureInfraCluster ensures an InfraCluster object exists in the cluster.
+func (r *InfraClusterController) ensureInfraCluster(ctx context.Context, log logr.Logger) (client.Object, error) {
+	var infraCluster client.Object
+	// TODO: implement InfraCluster generation for missing platforms.
```

Comment:
Can we remove this TODO now?

Reply:
Ideally we need to also support the other platforms with an ensure (aside from Openstack, which handles the creation separately in their own controller).